### PR TITLE
Remove ANIs and start splitting runtime

### PR DESCRIPTION
**Pull request by anurse**
* ANIs removed and converted to standard interfaces (this will break the world, @pranavkm and I will go around a fix it :))
* Split runtime at AppHost to prepare for some major refactoring - Right now the "new" runtime doesn't really exist :)

[There's a minor change coming to add something to add M.F.R.Interfaces to the dependencies copied in to the kre folder, stand by for that]

### CODE REVIEW

File: src/kre.clr.managed/kre.clr.managed.kproj [hunk 1]
```diff
@@ -13,6 +13,9 @@
   <PropertyGroup>
     <SchemaVersion>2.0</SchemaVersion>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (pranavkm) [inline src/kre.clr.managed/kre.clr.managed.kproj]: Intentional?

**Comment by pranavkm** [comment]
:shipit:

File: src/kre.clr.managed/kre.clr.managed.kproj (hunk 1 above)
Comment (anurse) [inline src/kre.clr.managed/kre.clr.managed.kproj]: No... VS did it :(. I'll remove.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**